### PR TITLE
27.1

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

File: src/main/java/internet/shop/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDb = userService.findByLogin(login)
+                .orElseThrow(() -> new AuthenticationException("Incorrect username or password"));
```
Comment (AnaLanda) [inline src/main/java/internet/shop/security/AuthenticationServiceImpl.java:17]:
```suggestion
        User userFromDb = userService.findByLogin(login).orElseThrow(() -> 
        new AuthenticationException("Incorrect username or password"));
```

**Review by AnaLanda — APPROVED** [review]
Looks good!

File: src/main/java/internet/shop/security/AuthenticationServiceImpl.java (hunk 1 above)
Comment (jah701) [reply src/main/java/internet/shop/security/AuthenticationServiceImpl.java:17]: @AnaLanda orElseThrow() по дефолту бросает NoSuchElementException(), по-этому, думаю, тут нам нужно выбросить именно наш :)

File: src/main/java/internet/shop/controller/order/CreateOrderController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import internet.shop.model.ShoppingCart;
+import internet.shop.service.OrderService;
+import internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CreateOrderController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/order/CreateOrderController.java:14]: this controller was in your previous PR(which you didn't merge) ,so it should not be in this one

File: src/main/java/internet/shop/controller/order/DeleteOrderController.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import internet.shop.lib.Injector;
+import internet.shop.service.OrderService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteOrderController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/order/DeleteOrderController.java:11]: same

File: src/main/java/internet/shop/controller/order/GetAllOrdersController.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import internet.shop.model.Order;
+import internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllOrdersController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/order/GetAllOrdersController.java:13]: same

File: src/main/java/internet/shop/controller/order/GetOrderDetailsController.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import internet.shop.lib.Injector;
+import internet.shop.model.Order;
+import internet.shop.service.OrderService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetOrderDetailsController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/order/GetOrderDetailsController.java:12]: same

File: src/main/java/internet/shop/controller/order/GetUserOrdersController.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import internet.shop.model.Order;
+import internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetUserOrdersController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/order/GetUserOrdersController.java:13]: same

File: src/main/java/internet/shop/controller/product/DeleteProductController.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+import internet.shop.lib.Injector;
+import internet.shop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteProductController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/product/DeleteProductController.java:11]: same

File: src/main/java/internet/shop/controller/product/ManageAllProductsController.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import internet.shop.model.Product;
+import internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ManageAllProductsController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/product/ManageAllProductsController.java:13]: same

File: src/main/java/internet/shop/controller/user/LoginController.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("pass");
+        try {
+            User user = authenticationService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_id", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/controller/user/LoginController.java:33]: let's make userId a constant

File: src/main/java/internet/shop/security/AuthenticationServiceImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/internet/shop/security/AuthenticationServiceImpl.java:17]: I think that @AnaLanda meant just code formatting

File: src/main/java/internet/shop/security/AuthenticationServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+        if (userFromDb.getPassword().equals(password)) {
+            return userFromDb;
+        }
+        throw new AuthenticationException("Incorrect username or password");
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/security/AuthenticationServiceImpl.java:21]: either make both checks in one if() or make exception message a constant

File: src/main/java/internet/shop/web/filters/AuthenticationFilter.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("internet.shop");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/shop/web/filters/AuthenticationFilter.java:21]:
```suggestion
    private static final Injector injector = Injector.getInstance("internet.shop");
    private UserService userService = (UserService) injector.getInstance(UserService.class);

    @Override
    public void init(FilterConfig filterConfig) throws ServletException {
```

**Review by dmytro-hryhoruk — COMMENTED** [review]
Remove all changes from this PR which are related to previous one

File: src/main/java/internet/shop/security/AuthenticationServiceImpl.java (hunk 1 above)
Comment (jah701) [reply src/main/java/internet/shop/security/AuthenticationServiceImpl.java:17]: oh.. :D

**Pull request closed without merging**